Core.Arango.Tests/Core/TestBase.cs

```diff
@@ -7,29 +7,148 @@
+        private const string ARANGODB_TOPOLOGY_ENVAR = "ARANGODB_TOPOLOGY";
+        private const string DefaultImage = "arangodb:latest";
+        private const string DefaultImagePassword = "password";
+        private const string DefaultImageUser = "root";
+        private const string ClusterValue = "cluster";
+
         public IArangoContext Arango { get; protected set; }
+        private readonly static Lazy<Task<IEnumerable<ArangoDbContainer>>> Containers = new(async () => 
+        Environment.GetEnvironmentVariable(ARANGODB_TOPOLOGY_ENVAR) == ClusterValue
+            ? await SetupClusterServer()
+            : await SetupSingleServer());
 
-        public virtual Task InitializeAsync()
+        public virtual async Task InitializeAsync()
         {
-            return Task.CompletedTask;
+            foreach (var container in await Containers.Value)
+            {
+                await container.StartAsync()
+                    .ConfigureAwait(false);
+            }
         }
 
-        public async Task DisposeAsync()
+        private static (Dictionary<string, string> Environment, string Version) GetVersionAndEnvironment()
         {
-            try
+            var environment = new Dictionary<string, string>();
+
+            if (string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(ARANGO_LICENSE_KEY_ENVAR)))
             {
-                foreach (var db in await Arango.Database.ListAsync())
-                    await Arango.Database.DropAsync(db);
+                environment.Add(ARANGO_LICENSE_KEY_ENVAR, Environment.GetEnvironmentVariable(ARANGO_LICENSE_KEY_ENVAR));
             }
-            catch
+
+            var version = string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(ARANGODB_VERSION_ENVAR))
+                ? DefaultImage
+                : Environment.GetEnvironmentVariable(ARANGODB_VERSION_ENVAR);
+
+            return (environment, version);
+        }
+
+        private static Task<List<ArangoDbContainer>> SetupSingleServer()
+        {
+            var (environment, version) = GetVersionAndEnvironment();
+            return Task.FromResult(new List<ArangoDbContainer>{new ArangoDbBuilder()
+                        .WithImage(version)
+                        .WithEnvironment(environment)
+                        .WithPassword(DefaultImagePassword)
+                        .Build() });
+        }
+
+        private static async Task<List<ArangoDbContainer>> SetupClusterServer()
+        {
+            var (environment, version) = GetVersionAndEnvironment();
+            
+            // This is not yet tested. But is a largely compatible replication of the start_db_cluster script.
+            // 2024-10-29 20:11:32 Error while processing command-line options for arangod:
+            // 2024 - 10 - 29 20:11:32   unknown option '--cluster.start-dbserver false'
+            // I suspect that's because Cluster is only available for Enterprise?
```

Comment:
Take note of this. 
This option is indeed not available in the main image of ArangoDB for `arangod`.